internal/lifecycle/lifecycle.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lifecycle
+
+import (
+	"fmt"
+	"github.com/GoogleCloudPlatform/serverless-sample-tester/internal/util"
+	"os"
+	"os/exec"
+)
+
+// Lifecycle is a list of ordered exec.Cmd that should be run to execute a certain process.
+type Lifecycle []*exec.Cmd
```

Comment:
Think I commented on this before but why use the indirection of a pointer here when you could just use a slice of exec.Cmd directly?

Reply:
Per an offline discussion with you (@tritone), we're planning on leaving this as is for now. Mainly because the Go SDK `exec.Command` function returns an `*exec.Cmd`pointer — instead of an `exec.Cmd` itself.